garrysmod/lua/vgui/ddialogbox.lua

```diff
@@ -0,0 +1,203 @@
+
+local PANEL = {}
+
+-- See: https://en.wikipedia.org/wiki/Dialog_box
+
+function PANEL:Init()
+	self:SetTitle( "Untitled Dialog Box" )
+	self:SetSize( 400, 200 )
+	self:MakePopup()
+	self:Center()
+	self:SetDrawOnTop( true )
+	
+	self.children = {}
+	
+	---
+	
+	self:SetText( "" )
+	
+	self:SetAcceptButtonText( "#GameUI_OK" )
+	self:SetAcceptButtonEnabled( true )
+	
+	self:SetRejectButtonText( "#GameUI_Cancel" )
+	self:SetRejectButtonEnabled( true )
+end
+
+function PANEL:GenerateExample( className, propertySheet, width, height )
+	local dialog = vgui.Create( className )
+	dialog:SetIcon( "icon16/comment.png" )
+	dialog:SetTitle( "Example Dialog" )
+	dialog:SetText( "This is an example dialog box.\n\nYou can display information to the user here and request an action be taken.\n" )
+	dialog:SetAcceptButtonText( "Custom Accept Button Text" )
+	
+	local firstNameInput = dialog:Add( "DTextEntry" )
+	firstNameInput:SetPlaceholderText( "First name" )
+	firstNameInput:DockMargin( 0, 0, 0, 0 )
+	
+	local lastNameInput = dialog:Add( "DTextEntry" )
+	lastNameInput:SetPlaceholderText( "Last name" )
+	lastNameInput:DockMargin( 0, 4, 0, 0 )
+	lastNameInput.OnEnter = function()
+		dialog:OnAccept()
+		dialog:Close()
+	end
+
+	propertySheet:AddSheet( className, dialog, nil, true, true )
+end
+
+function PANEL:OnClose()
+	if not self.handled then
```

Comment:
Code formatting is not compliant with [CONTRIBUTING.md](https://github.com/Facepunch/garrysmod/blob/master/CONTRIBUTING.md)

Specifically missing brackets around the `if-` block. There are more instances that I have not pointed out.